detect/directory.go

```diff
@@ -49,64 +50,33 @@ func (d *Detector) DetectFiles(paths <-chan sources.ScanTarget) ([]report.Findin
 				}
 			}
 
-			// Buffer to hold file chunks
-			buf := make([]byte, chunkSize)
-			totalLines := 0
+			var (
+				// Buffer to hold file chunks
+				reader     = bufio.NewReader(f)
```

Comment:
Using #1432 as a test case. @ahrav @zricethezav any obvious errors with my use of `bufio.Reader`?

**Before**
```
12:33PM INF scanned ~104857630 bytes (104.86 MB) in 5.63s
12:33PM WRN leaks found: 1
```
**After**
```
1:05PM INF scanned ~104857630 bytes (104.86 MB) in 1.42s
1:05PM WRN leaks found: 1
```

Reply:
This looks correct to me. I agree—it does feel a bit odd to use all three functions, but if I’m understanding this correctly, they each serve different purposes, so it actually makes sense.

Reply:
I'm concerned that `f.Read` and `r.ReadByte` may not place nicely. E.g., after the first iteration does calling `f.Read` also progress the internal reader used by bufio?

The number of bytes read (from #1670) appears to be consistent. 🤷